qiita_db/support_files/patches/python_patches/66.py

```diff
@@ -1,15 +1,26 @@
-# August 6, 2018
-# Create parameters for the ssh/scp remote file upload commands
-
-
 from json import loads, dumps
-
 from qiita_db.sql_connection import TRN
 from qiita_db.software import Software, Command
 from qiita_db.exceptions import (QiitaDBError, QiitaDBDuplicateError)
 from qiita_db.util import convert_to_id
+from qiita_db.study import Study
+from re import sub
 
+# August 31, 2018
+# Strip any UTF-8 characters that are not also printable ASCII characters
+# from study titles. As some analysis packages cannot interpret UTF-8
+# characters, it becomes important to remove them from study titles, as
+# they are used as metadata/identifiers when creating new analyses.
+studies = Study.get_by_status('public')
```

Comment:
Just realized that this will only loop over the public studies and we need to check _all_ studies ...

Reply:
Looking through the methods, I haven't yet seen a way to enumerate through all of the studies, regardless of status (Study.get_all() or similar). It looks like the source of truth for the set of valid status strings is ultimately the qiita.visibility table, which lists the following:
   awaiting_approval
   sandbox
   private
   public

A nested loop to process each status in turn isn't horrible, but this might be a good time to write a get_all(), if it's not already available.

Another alternative would be to simply perform the same action in SQL directly. It looks like the only state we're changing through the API is study_title in the qiita.study table. The update statement below has already been tested, and could be placed in 66.sql:

update qiita.study set study_title = REGEXP_REPLACE(study_title,'[^\x20-\x7E]+','','g');

What do you think?

Reply:
Good point, perhaps we should open an issue about making sure that all objects have an __iter__ method. Anyway, we normally do:
```
studies = Study.get_by_status('private').union(
    Study.get_by_status('public')).union(Study.get_by_status('sandbox'))
```